npm_and_yarn/lib/dependabot/npm_and_yarn/helpers.rb

```diff
@@ -453,7 +471,47 @@ def self.fallback_to_local_version(name)
         package_manager_activate(name, current_version)
       end
 
-      # Install the package manager for specified version by using corepack
+      # Find cached version that matches the requested version pattern
+      sig { params(name: T.deprecated_enum(%w(yarn pnpm npm)), version: String).returns(T.nilable(String)) }
```

Comment:
Is using `deprecated_enum` better than what I suggested? As the name suggests, this pattern is deprecated and it may no longer be recommended